resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/filesystem/zip/CopyTest.java

```diff
@@ -0,0 +1,211 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors: Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.core.tests.filesystem.zip;
+
+import static org.eclipse.core.tests.filesystem.zip.ZipFileSystemTestUtil.assertTextFileContent;
+import static org.eclipse.core.tests.filesystem.zip.ZipFileSystemTestUtil.ensureDoesNotExist;
+import static org.eclipse.core.tests.filesystem.zip.ZipFileSystemTestUtil.ensureExists;
+import static org.eclipse.core.tests.filesystem.zip.ZipFileSystemTestUtil.getMonitor;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ *
+ */
+public class CopyTest {
+
+	@BeforeEach
+	public void setup() throws Exception {
+		ZipFileSystemTestSetup.defaultSetup();
+	}
+
+	@AfterEach
+	public void teardown() throws Exception {
+		ZipFileSystemTestSetup.teardown();
+	}
+
+	@ParameterizedTest
```

Comment:
Avoid "@ParameterizedTest" as long as you do not really need it as it makes trouble building a good test name within the build jobs